esp/services/ws_dali/ws_daliservice.cpp

```diff
@@ -0,0 +1,136 @@
+    if ((strstr(path, "::") != nullptr) && !strchr(path, '/'))
+    {
+        lfn.set(path);
+        lfn.makeFullnameQuery(lfnPath, DXB_File);
+        path = lfnPath.str();
+    }
+    else if (strchr(path + ((*path == '/') ? 1 : 0),'/') == nullptr)
+        safe = true;    // all root trees safe
+
+    Owned<IRemoteConnection> conn = querySDS().connect(remLeading(path), myProcessSession(), safe ? 0 : RTM_LOCK_READ, daliConnectTimeoutMs);
+    if (!conn && !lfnPath.isEmpty())
+    {
+        lfn.makeFullnameQuery(lfnPath.clear(), DXB_SuperFile);
+        path = lfnPath.str();
+        conn.setown(querySDS().connect(remLeading(path), myProcessSession(), safe? 0 : RTM_LOCK_READ, daliConnectTimeoutMs));
+    }
+    if (conn.get())
+        xpath.append(path);
+    return conn.getClear();
+}
+
+void CWSDaliEx::init(IPropertyTree* cfg, const char* process, const char* service)
+{
+    espProcess.set(process);
+}
+
+int CWSDaliSoapBindingEx::onGet(CHttpRequest* request, CHttpResponse* response)
+{
+    try
+    {
+#ifdef _USE_OPENLDAP
+        request->queryContext()->ensureSuperUser(ECLWATCH_SUPER_USER_ACCESS_DENIED, "Access denied, administrators only.");
+#endif
+        if (wsdService->isDaliDetached())
+            throw MakeStringException(ECLWATCH_CANNOT_CONNECT_DALI, "Dali detached.");
+
+        StringBuffer path;
+        request->getPath(path);
+
+        if (!strnicmp(path.str(), REQPATH_EXPORTSDSDATA, sizeof(REQPATH_EXPORTSDSDATA) - 1))
+        {
+            exportSDSData(request, response);
+            return 0;
+        }
+    }
+    catch(IException* e)
+    {
+        onGetException(*request->queryContext(), request, response, *e);
+        FORWARDEXCEPTION(*request->queryContext(), e,  ECLWATCH_INTERNAL_ERROR);
+    }
+
+    return CWSDaliSoapBinding::onGet(request,response);
+}
+
+void CWSDaliSoapBindingEx::exportSDSData(CHttpRequest* request, CHttpResponse* response)
+{
+    CDateTime now;
+    now.setNow();
+    time_t createTime = now.getSimple();
+
+    StringBuffer peer, outFileNameWithPath;
+    VStringBuffer outFileName("sds_for_%s_%ld", request->getPeer(peer).str(), createTime);
+    outFileNameWithPath.set(daliFolder).append(outFileName);
+
+    StringBuffer path, xpath, safeReq;
+    request->getParameter("Path", path);
+    request->getParameter("Safe", safeReq);
+    bool safe = atoi(safeReq.str()) ? true : false;
+    Owned<IRemoteConnection> conn = connectXPathOrFile(path, safe, xpath);
+    if (!conn)
+        throw MakeStringException(ECLWATCH_CANNOT_CONNECT_DALI, "Failed to connect Dali.");
+
+    Owned<IFile> workingDir = createIFile(daliFolder);
+    if (!workingDir->exists())
+        workingDir->createDirectory();
+
+    Owned<IPropertyTree> root = conn->getRoot();
+    Owned<IFile> f = createIFile(outFileNameWithPath);
+    Owned<IFileIO> io = f->open(IFOcreaterw);
+    { //Force the fios to finish
+        Owned<IFileIOStream> fios = createBufferedIOStream(io);
+        toXML(root, *fios);
```

Comment:
I'm not sure why it's being exported to a file (as opposed to the content being returned), and/or how the client would get to it once it had been written to a file on the server ?

Reply:
As I said in the previous comment, I am using file streaming because the content of the export (whole SDS tree) could be huge.

Reply:
If using a file you could consider using the function 
```
extern jlib_decl IFileIO *createUniqueFile(const char *dir, const char *prefix, const char *ext, StringBuffer &tmpName);
```
or similar.

Reply:
I see you've opened a JIRA related to being able to stream this result directly in the future.
HPCC-25391 "Support streaming a PTree content to ESP response".